traffic-advice.bs

```diff
@@ -0,0 +1,265 @@
+        <div class="note">If present, the [[HTTP-SEMANTICS]] `Retry-After` response header could be used as a hint about when to next retry.</div>
+
+    1.  If |response|'s [=response/status=] is not an [=ok status=], then terminate the fetch, run |whenComplete| with null and return.
+
+    1.  If |response|'s [=response/status=] is a [=null body status=], then terminate the fetch, run |whenComplete| with null and return.
+
+    1.  Let |mimeType| be the result of [=header list/extracting a MIME type=] from |response|'s [=response/header list=].
+
+    1.  If |mimeType| is failure or its [=MIME type/essence=] is not `"application/trafficadvice+json"`, then terminate the fetch, run |whenComplete| with null and return.
+
+1.  Let |processResponseEndOfBody| be the following steps, given [=response=] |response| and null, failure or [=byte sequence=] |body|:
+
+    1.  If |body| is not a [=byte sequence=], then run |whenComplete| with null and return.
+
+    1.  Let |string| be the result of [=UTF-8 decoding=] |body|.
+
+    1.  Let |parseResult| be the result of [=parsing traffic advice=] from |string| given |identity|.
+
+    1.  Run |whenComplete| with |parseResult|.
+
+1.  [=Fetch=] |request| with |processResponse| and |processResponseEndOfBody|.
+
+</section>
+
+Parsing {#parsing}
+==================
+
+<section algorithm="parse-traffic-advice">
+
+To <dfn>parse traffic advice</dfn> from a [=string=] |string| given [=agent identity=] |identity|:
+
+1.  Let |parsed| be the result of [=parsing JSON into Infra values=] given |string|. If this throws an exception, then return null.
+
+1.  If |parsed| is not a [=list=], then return null.
+
+1.  Let |bestMatch| be null.
+
+1.  [=list/For each=] |entry| of |parsed|:
+
+    1.  If |entry| is not a [=map=], then [=iteration/continue=].
+
+    1.  If |entry|[`"user_agent"`] does not [=map/exist=] or is not a [=string=], then [=iteration/continue=].
+
+    1.  Let |agentSelector| be |entry|[`"user_agent"`].
+
+    1.  If |identity| does not contain |agentSelector|, then [=iteration/continue=].
+
+    1.  If |bestMatch| is null or |agentSelector| appears at an earlier index in |identity| than |bestMatch|[`"user_agent"`], then set |bestMatch| to |entry|.
+
+1.  If |bestMatch| is null, then return null.
+
+1.  Let |entry| be a [=traffic advice entry=].
+
+1.  If |bestMatch|[`"disallowed"`] [=map/exists=] and is true, then set |entry|'s [=traffic advice entry/disallowed flag=] to true.
+
+1.  Return |entry|.
+
+</section>
+
+Interpretation {#interpretation}
+================================
+
+*This section is non-normative.*
+
+[=Traffic advice respecting agents=] should, when they would be able to respect advice to disallow traffic to an origin (for example, when requested to proxy prefetch traffic to the origin), [=fetch traffic advice=] (respecting HTTP caching semantics).
```

Comment:
Avoid "should" and other such words in non-normative sections.

This section seems relatively normative to me.

Reply:
I can remove the non-normative line if you'd like but it's kinda weird as normative text because this document doesn't really aim to specify exactly how prefetch proxies (or other implementers) react to the advice, but I wanted to give it some context. Would you rather I relabel it as normative, replace "should" with some synonym (everything I tried seemed awkward), or something else.

Reply:
To be it seems better relabeled as normative. It feels about as normative as many HTTP RFCs I read when they try to specify client or server behavior :).